doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -264,16 +264,19 @@ SQL objects that don't exist in the compute layer (such as views) are omitted.
 
 ## `mz_compute_hydration_statuses`
 
-The `mz_compute_hydration_statuses` table describes the per-replica hydration status of each compute object (index, materialized view, or subscription).
+The `mz_compute_hydration_statuses` view describes the per-replica hydration status of each compute object (index, materialized view).
 
 A compute object is hydrated on a given replica when it has fully processed the initial snapshot of data available in its inputs.
 
 <!-- RELATION_SPEC mz_internal.mz_compute_hydration_statuses -->
-| Field        | Type        | Meaning  |
-| -----------  | ----------- | -------- |
-| `object_id`  | [`text`]    | The ID of a compute object. Corresponds to [`mz_catalog.mz_indexes.id`](../mz_catalog#mz_indexes), [`mz_catalog.mz_materialized_views.id`](../mz_catalog#mz_materialized_views), or [`mz_internal.mz_subscriptions`](#mz_subscriptions). |
-| `replica_id` | [`text`]    | The ID of a cluster replica. |
-| `hydrated`   | [`boolean`] | Whether the compute object is hydrated on the replica. |
+| Field            | Type         | Meaning  |
+| ---------------- | ------------ | -------- |
+| `object_id`      | [`text`]     | The ID of a compute object. Corresponds to [`mz_catalog.mz_indexes.id`](../mz_catalog#mz_indexes) or [`mz_catalog.mz_materialized_views.id`](../mz_catalog#mz_materialized_views) |
+| `replica_id`     | [`text`]     | The ID of a cluster replica. |
+| `hydrated`       | [`boolean`]  | Whether the compute object is hydrated on the replica. |
+| `hydration_time` | [`interval`] | The amount of time it took for the replica to hydrate the compute object. |
```

Comment:
Do we need the separate `hydrated` boolean if it is just `time_ns IS NOT NULL`?

Reply:
I think we need it for backward-compatibility because the [blue/green readiness query](https://github.com/teskje/materialize/blob/48d2ef6a1195601b7c3ce68945939de5d0410632/test/cluster/blue-green-deployment/setup.td#L69-L101) relies on it. I think most users run this query through our dbt adapter, but some might run it directly, and we should avoid breaking their workflows if we can. That's also why I'm including MVs with empty frontiers here.

We should probably make the blue/green readiness query a builtin view, or provide another way to query the same information, so we don't have to quasi-stabilize some of our `mz_internal` relations.

Reply:
Theoretically there should be no backward-compatibility issues with `mz_internal` objects. If people are relying on this object then I think we should promote it to `mz_catalog` or create a stable view as you suggest.

Reply:
It might even be the case that no users are using the b/g query directly, but the dbt adapter still does. We can of course change the query that uses, but I don't think we can control the cadence at which users update the dbt adapter, so we'd at least have to keep the old column around for some time.

But I think it's actually not bad UX-wise to have the additional column. Saves people that only care about whether an object is hydrated or not from having to interpret the time value (or the absence of it).

Reply:
Yeah, I think this change is fine.

Reply:
Yup, that sounds good!

Reply:
> I think most users run this query through our dbt adapter, but some might run it directly, and we should avoid breaking their workflows if we can.

These queries aren't exposed to users, and users are also discouraged from scripting their own orchestration for blue/green deployments.

> We can of course change the query that uses, but I don't think we can control the cadence at which users update the dbt adapter, so we'd at least have to keep the old column around for some time.

Users are typically encouraged to update when there's a new minor release, but less so if there's a patch release (unless it fixes a specific issue they're waiting for). dbt does depend **a lot** on introspection, though, so we should discuss how to deal with breaking changes to system catalog tables that affect it (or Terraform).